nautobot/ipam/views.py

```diff
@@ -672,7 +672,12 @@ class PrefixBulkDeleteView(generic.BulkDeleteView):
 
 
 class IPAddressListView(generic.ObjectListView):
-    queryset = IPAddress.objects.select_related("tenant", "status", "role")
+    queryset = IPAddress.objects.select_related("tenant", "status", "role").annotate(
+        interface_count=Count("interfaces"),
+        interface_parent_count=(Count("interfaces__device", distinct=True)),
+        vm_interface_count=Count("vm_interfaces"),
+        vm_interface_parent_count=(Count("vm_interfaces__virtual_machine", distinct=True)),
+    )
```

Comment:
Is this efficient enough as is or do we need to add anything to `select_related`/`prefetch_related`?

Reply:
I added `.prefetch_related("interfaces", "vm_interfaces")` to this query and didn't see any changes in the queries for listing this view.